backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt

```diff
@@ -1322,23 +1322,25 @@ internal class CodeGeneratorVisitor(val context: Context, val lifetimes: Map<IrE
 
     //-------------------------------------------------------------------------//
 
-    private fun evaluateTypeOperator(value: IrTypeOperatorCall): LLVMValueRef {
-        return when (value.operator) {
-            IrTypeOperator.CAST                      -> evaluateCast(value)
-            IrTypeOperator.IMPLICIT_INTEGER_COERCION -> evaluateIntegerCoercion(value)
-            IrTypeOperator.IMPLICIT_CAST             -> evaluateExpression(value.argument)
-            IrTypeOperator.IMPLICIT_NOTNULL          -> TODO(ir2string(value))
-            IrTypeOperator.IMPLICIT_COERCION_TO_UNIT -> {
-                evaluateExpression(value.argument)
-                functionGenerationContext.theUnitInstanceRef.llvm
-            }
-            IrTypeOperator.SAFE_CAST                 -> throw IllegalStateException("safe cast wasn't lowered")
-            IrTypeOperator.INSTANCEOF                -> evaluateInstanceOf(value)
-            IrTypeOperator.NOT_INSTANCEOF            -> evaluateNotInstanceOf(value)
-            IrTypeOperator.SAM_CONVERSION            -> TODO(ir2string(value))
-            IrTypeOperator.IMPLICIT_DYNAMIC_CAST     -> TODO(ir2string(value))
-            IrTypeOperator.REINTERPRET_CAST          -> TODO(ir2string(value))
+    private fun evaluateTypeOperator(value: IrTypeOperatorCall) = when (value.operator) {
+        IrTypeOperator.IMPLICIT_INTEGER_COERCION -> evaluateIntegerCoercion(value)
+        IrTypeOperator.IMPLICIT_CAST             -> evaluateExpression(value.argument)
+        IrTypeOperator.IMPLICIT_COERCION_TO_UNIT -> {
+            evaluateExpression(value.argument)
+            functionGenerationContext.theUnitInstanceRef.llvm
         }
+
+        IrTypeOperator.CAST,
+        IrTypeOperator.SAFE_CAST,
+        IrTypeOperator.INSTANCEOF,
+        IrTypeOperator.NOT_INSTANCEOF ->
+            error("Should've been lowered: ${value.dump()} ${functionGenerationContext.irFunction?.render()}")
```

Comment:
You can report the exact source location here too, see https://github.com/JetBrains/kotlin-native/blob/5cb1de5d84392f6d1bb836f299fc6fbc8bbe962d/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/Reporting.kt#L34